### PR TITLE
Respect test selectors in TestNGRunner

### PR DESCRIPTION
**Pull request by jiangty-addepar**
On top of #1707: actually handle test selectors when running TestNG tests.

### CODE REVIEW

File: .idea/codeStyleSettings.xml [hunk 1]
```diff
@@ -446,6 +446,5 @@
       </value>
     </option>
     <option name="USE_PER_PROJECT_SETTINGS" value="true" />
-    <option name="PREFERRED_PROJECT_CODE_STYLE" value="GoogleStyle" />
```
Comment (ttsugriy) [inline .idea/codeStyleSettings.xml]: please revert this change

Comment (jiangty-addepar) [reply .idea/codeStyleSettings.xml]: oops, sorry!

**Comment by jiangty-addepar** [comment]
heh, trying to commit from a non-work computer caused some problems---sorry!

**Comment by ttsugriy** [comment]
looks like github is now complaining about a merge conflict...

**Comment by jiangty-addepar** [comment]
@ttsugriy oh, I think this is because the previous PR was merged; fixing

**Comment by jiangty-addepar** [comment]
this and #1707 should be fixed now
